### PR TITLE
Added new iterative lookup algo

### PR DESCRIPTION
**Pull request by kag0**
Added cap on SocketIOWrapper reads
Moved all test files into test folder and respective packages
Added new blackdoor jar

### CODE REVIEW

File: src/blackdoor/cqbe/output_logic/Router.java [hunk 1]
```diff
@@ -339,47 +337,113 @@ public static AddressTable primitiveLookup(L3Address remoteNode, Address destina
 		return ret;
 	}
 	
+	public AddressTable iterativeLookup(Address destination, int α, int n){
+		return iterativeLookup(destination, α, n, AddressTable.DEFAULT_MAX_SIZE * 2);
+	}
+	
+	public AddressTable iterativeLookup(Address destination, int α, int n, int Ω){
```
Comment (CjBuresch) [inline src/blackdoor/cqbe/output_logic/Router.java]: Why the special characters? not that its a huge deal. just curious

**Comment by CjBuresch** [comment]
Looks good.

File: src/blackdoor/cqbe/output_logic/Router.java (hunk 1 above)
Comment (euphzji) [inline src/blackdoor/cqbe/output_logic/Router.java]: What do they signify?

Comment (kag0) [inline src/blackdoor/cqbe/output_logic/Router.java]: If you look at the protocol doc it's in section 5.2

**Pull request merged**